neo/Consensus/ConsensusService.cs

```diff
@@ -212,13 +245,12 @@ private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest m
             context.NextConsensus = message.NextConsensus;
             context.TransactionHashes = message.TransactionHashes;
             context.Transactions = new Dictionary<UInt256, Transaction>();
+            context.Preparations[payload.ValidatorIndex] = true;
             byte[] hashData = context.MakeHeader().GetHashData();
-            if (!Crypto.Default.VerifySignature(hashData, message.Signature, context.Validators[payload.ValidatorIndex].EncodePoint(false))) return;
-            for (int i = 0; i < context.Signatures.Length; i++)
-                if (context.Signatures[i] != null)
-                    if (!Crypto.Default.VerifySignature(hashData, context.Signatures[i], context.Validators[i].EncodePoint(false)))
-                        context.Signatures[i] = null;
-            context.Signatures[payload.ValidatorIndex] = message.Signature;
+            for (int i = 0; i < context.Commits.Length; i++)
+                if (context.Commits[i] != null)
+                    if (!Crypto.Default.VerifySignature(hashData, context.Commits[i], context.Validators[i].EncodePoint(false)))
```

Comment:
probably, the HashData will be null at this point, Erik.

Reply:
It is impossible. Because we have already received the `PrepareRequest`, so we have enough information to build the block header.